### PR TITLE
Fixes #24189 - use new foreman upgrade task

### PR DESCRIPTION
**Pull request by jlsherrill**
This removes all pre-3.7 upgrade tasks and migrates 3.7 
and newer to use the new upgrade rake task

Requires: https://github.com/theforeman/foreman/pull/5780

### CODE REVIEW

**Comment by ehelms** [comment]
Would this mean that any users that want to upgrade have to upgrade to 3.7 before upgrading to 3.8? I ask because our docs language typically doesn't discuss upgrading sequentially.

Will this re-run the upgrade steps for any user that upgrades to 3.7 when they upgrade to 3.8?

**Comment by jlsherrill** [comment]
> Would this mean that any users that want to upgrade have to upgrade to 3.7 before upgrading to 3.8? I ask because our docs language typically doesn't discuss upgrading sequentially.

I don't think this would mean that, is there something here that makes you think that?

> Will this re-run the upgrade steps for any user that upgrades to 3.7 when they upgrade to 3.8?
It will.  There was only 1 for 3.7 and it seemed fairly minor.  I the risk vs reward for is more in favor of just re-running that task rather than build a system to load up already run-tasks.  I'm open for debate on that though :)

**Comment by jlsherrill** [comment]
Actually, since this probably won't go in until 3.9, i'll just remove the one existing 3.7 upgrade task

File: db/seeds.d/111-upgrade_tasks.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+UpgradeTask.define_tasks(:katello) do
+  [
+    {:name => 'katello:correct_repositories', :long_running => true, :skip_failure => true, :always_run => true},
+    {:name => 'katello:correct_puppet_environments', :long_running => true, :skip_failure => true, :always_run => true},
+    {:name => 'katello:clean_backend_objects', :long_running => true, :skip_failure => true, :always_run => true},
+    {:name => 'katello:clean_backend_objects', :long_running => true, :skip_failure => true, :always_run => true},
+    {:name => 'katello:upgrades:3.7:make_all_ks_repos_bootable', :long_running => true}
```
Comment (akofink) [inline db/seeds.d/111-upgrade_tasks.rb]: for rake upgrades in the past, it looks like we were running all the upgrade tasks, even from 3.0. Is that true, and are they all designed to be idempotent? If so, why not just include all those tasks in the new plugin upgrades? Why just include the 3.7 one?

Comment (jlsherrill) [reply db/seeds.d/111-upgrade_tasks.rb]: They are designed to be idempotent, but most upgrade tasks are only run once.  The fact that those once-only tasks means that we were just keeping them around for no reason, in addition to that:

1.  they would get run on first upgrade, even if not needed (a katello 2.3 upgrade task would get run when installing a fresh katello 3.7 and then running --upgrade the first time).
2.  We don't support upgrading from such an older version to such a newer version.  In the past we've guided, we support upgrades from N-2, but we haven't even tested that, generally we only test N-1. 

Maybe we should add the deletion of old upgrade tasks to the release steps?

Comment (akofink) [reply db/seeds.d/111-upgrade_tasks.rb]: Hmm...  I'm ok with adding to the release process, if it's easy. What about a rake task to remove old upgrade tasks based on the VERSION?

Comment (jlsherrill) [reply db/seeds.d/111-upgrade_tasks.rb]: It would have to somehow edit this seeds file, which seems kinda weird.

Comment (jlsherrill) [reply db/seeds.d/111-upgrade_tasks.rb]: and actually, i meant to remove the 3.7 upgrade task since this won't make it till 3.9, let me do that.

Comment (akofink) [reply db/seeds.d/111-upgrade_tasks.rb]: If we add a trailing comma, it'd be easier to write automation in tool_belt to do this.

Comment (jlsherrill) [reply db/seeds.d/111-upgrade_tasks.rb]: added, and accidentally had clean_backend_objects in there twice, so i removed that :)

**Pull request merged**